workflow/controller/scope.go

```diff
@@ -38,23 +35,16 @@ func (s *wfScope) addArtifactToScope(key string, artifact wfv1.Artifact) {
 
 // resolveVar resolves a parameter or artifact
 func (s *wfScope) resolveVar(v string) (interface{}, error) {
-	v = strings.TrimPrefix(v, "{{")
-	v = strings.TrimSuffix(v, "}}")
-	parts := strings.Split(v, ".")
-	prefix := parts[0]
-	switch prefix {
-	case "steps", "tasks", "workflow":
-		val, ok := s.scope[v]
-		if ok {
-			return val, nil
-		}
-	case "inputs":
-		art := s.tmpl.Inputs.GetArtifactByName(parts[2])
-		if art != nil {
-			return *art, nil
+	m := make(map[string]interface{})
+	for k, v := range s.scope {
```

Comment:
s.scope is already `map[string]interface{}`. can we just add new keys to it?

Reply:
would modify it, want to copy